htlcswitch/link.go

```diff
@@ -2144,8 +2144,8 @@ func (l *channelLink) UpdateForwardingPolicy(newPolicy ForwardingPolicy) {
 // NOTE: Part of the ChannelLink interface.
 func (l *channelLink) CheckHtlcForward(payHash [32]byte,
 	incomingHtlcAmt, amtToForward lnwire.MilliSatoshi,
-	incomingTimeout, outgoingTimeout uint32,
-	heightNow uint32) *LinkError {
+	incomingTimeout, outgoingTimeout, heightNow uint32,
+	incomingLink lnwire.ShortChannelID, allowCircular bool) *LinkError {
```

Comment:
Maybe it is better to get this static config bool from the link configuration, rather than passing it every time.

Reply:
But I actually think that this check is something that should be done in the switch, when the forwarding decision is made.

Reply:
I thought about putting it in the switch (in `handlepacketforward`) but then it's less easy to unit test. It's quite a simple change though so can move it in there as well, which saves having to get that bool down to the link?